donate/payments/sqs.py

```diff
@@ -18,14 +24,40 @@ def sqs_client():
         )
 
 
+def attempt_send_to_sqs(client, payload):
+    client.send_message(
+        QueueUrl=settings.BASKET_SQS_QUEUE_URL,
+        MessageBody=json.dumps(payload, cls=DjangoJSONEncoder, sort_keys=True),
+    )
+
+
 def send_to_sqs(payload):
     # If BASKET_SQS_QUEUE_URL is not configured, do nothing (djangorq is logging the payload).
     if settings.BASKET_SQS_QUEUE_URL:
         client = sqs_client()
         if client is None:
+            logger.error("Could not connect to SQS Client.")
             return
 
-        return client.send_message(
-            QueueUrl=settings.BASKET_SQS_QUEUE_URL,
-            MessageBody=json.dumps(payload, cls=DjangoJSONEncoder, sort_keys=True),
-        )
+    send_retries = 3
+    send_data_immediately = 0  # seconds
+    send_data_delay = 10  # seconds
+    schedule_priority = 1
+    call_args = (client, payload)
+
+    for attempt in range(send_retries):
+        if attempt == 0:
+            schedule.enter(send_data_immediately, schedule_priority, attempt_send_to_sqs, call_args)
+        else:
+            schedule.enter(send_data_delay, schedule_priority, attempt_send_to_sqs, call_args)
+        try:
+            # explicitly block, just in case the implicit behaviour changes in the future
+            schedule.run(blocking=True)
+            break
+        except botocore.exceptions.ClientError as err:
+            if attempt != 2:
+                logger.error(f"Error when sending data to SQS: {err}")
```

Comment:
Hi @Pomax!

While working on the safe attribute access, I decided to see what `err` itself prints as. 

What I found was it actually displays a more detailed message that I think would be of use, in case the donate stack is having trouble connecting to SQS.

For example, here is an error message that was returned when I added a false queueURL to the sendmessage function:
`An error occurred (AWS.SimpleQueueService.NonExistentQueue) when calling the SendMessage operation: The specified queue does not exist for this wsdl version.`

I am now voting for printing the entire error instead of err.response['Error']['Message]' for the following reasons:
- Avoids the need to do a safe attribute check
- Prints where and why the error occured, which would be more helpful to people troubleshooting

Reply:
agreed, that's much better.